packages/affine/block-paragraph/src/commands/indent-paragraph.ts

```diff
@@ -80,9 +84,31 @@ export const indentParagraphCommand: Command<'indentContext'> = (ctx, next) => {
   if (!previousSibling) return;
 
   doc.captureSync();
-  doc.moveBlocks([model], previousSibling);
 
-  // update collapsed state
+  // update collapsed state of affine paragraph
+  const nearestHeading = getNearestHeadingBefore(model);
+  if (
+    nearestHeading &&
+    matchFlavours(nearestHeading, ['affine:paragraph']) &&
+    nearestHeading.collapsed
+  ) {
+    doc.updateBlock(nearestHeading, {
+      collapsed: false,
+    });
+  }
+
+  if (
+    matchFlavours(model, ['affine:paragraph']) &&
+    model.type.startsWith('h') &&
+    model.collapsed
+  ) {
+    const collapsedSiblings = calculateCollapsedSiblings(model);
+    doc.moveBlocks([model, ...collapsedSiblings], previousSibling);
+  } else {
+    doc.moveBlocks([model], previousSibling);
```

Comment:
I think `getNearestHeadingBefore` and updating collapsed state should be after indenting paragraph. There is a case show in the following video

https://github.com/user-attachments/assets/4e0ef45a-d3e2-436f-b867-b14dfd7f3cef

Reply:
I didn't consider this case; it seems that the collapse state needs to be updated both before and after moving lol.

https://github.com/user-attachments/assets/25c61db4-26a7-44cc-8f5a-8cd073f60d2b